server/channels/store/sqlstore/shared_channel_store.go

```diff
@@ -761,45 +761,38 @@ func (s SqlSharedChannelStore) GetUsersForSync(filter model.GetUsersForSyncFilte
 // UpdateUserLastSyncAt updates the LastSyncAt timestamp for the specified SharedChannelUser.
 func (s SqlSharedChannelStore) UpdateUserLastSyncAt(userID string, channelID string, remoteID string) error {
 	var query string
+	// squirrel cannot handle the difference in syntax for update with join
 	if s.DriverName() == model.DatabaseDriverPostgres {
 		query = `
 		UPDATE
 			SharedChannelUsers AS scu
 		SET
-			LastSyncAt = GREATEST(Users.UpdateAt, Users.LastPictureUpdate)
+			LastSyncAt = GREATEST(u.UpdateAt, u.LastPictureUpdate)
 		FROM
-			Users
+			Users AS u
 		WHERE
-			Users.Id = scu.UserId AND scu.UserId = ? AND scu.ChannelId = ? AND scu.RemoteId = ?
+			u.Id = scu.UserId AND scu.UserId = ? AND scu.ChannelId = ? AND scu.RemoteId = ?
 		`
 	} else if s.DriverName() == model.DatabaseDriverMysql {
 		query = `
 		UPDATE
 			SharedChannelUsers AS scu
 		INNER JOIN
-			Users ON scu.UserId = Users.Id
+			Users AS u ON u.Id = scu.UserId
 		SET
-			LastSyncAt = GREATEST(Users.UpdateAt, Users.LastPictureUpdate)
+			LastSyncAt = GREATEST(u.UpdateAt, u.LastPictureUpdate)
 		WHERE
 			scu.UserId = ? AND scu.ChannelId = ? AND scu.RemoteId = ?
 		`
 	} else {
 		return errors.New("unsupported DB driver " + s.DriverName())
 	}
 
-	result, err := s.GetMasterX().Exec(query, userID, channelID, remoteID)
+	_, err := s.GetMasterX().Exec(query, userID, channelID, remoteID)
 	if err != nil {
 		return fmt.Errorf("failed to update LastSyncAt for SharedChannelUser with userId=%s, channelId=%s, remoteId=%s: %w",
 			userID, channelID, remoteID, err)
 	}
-
-	count, err := result.RowsAffected()
```

Comment:
This deleted because RowsAffected is only non-zero if a column's value is actually changed, at least for MySQL.  This can be called where LastSyncAt will be set to its current value, with a result of zero row affected.  We no longer consider that an error to avoid logging meaningless errors.